### DISCUSSION TITLE
Styling Not Being Applied in NextJS-13

### DISCUSSION THREAD

**Discussion by MattHeslington** [tailwindlabs/tailwindcss#9936] [Help]
Hello,

I'm experiencing a really strange issue with the experimental version of NextJs13 whereby styling is not being applied despite the rules being present in the static stylesheet. I have tried both the TS and JS implementations of Next, and have tried different browsers too.

I've folowed both the instructions on the Tailwind site, and those on the Next site for implementing Tailwind. Nothing has worked correctly.

Here we can see a div with 3 Tailwind classes, but only font-bold is being honoured:

![Screenshot 2022-11-25 at 19 17 11](https://user-images.githubusercontent.com/8208896/203975210-7c8343fa-8a51-42a1-af41-4448d7a99948.jpg)

I thought that maybe there was an issue with the classes not being compiled, but looking at the static stylesheet, they are there:

![Screenshot 2022-11-25 at 19 16 29](https://user-images.githubusercontent.com/8208896/203975444-53ab2f1f-347c-4749-99cc-09f5fd56704f.jpg)

The only way I can get the classes to show is to manually reload or refresh the browser, but obviously this isn' t ideal in a large app.

I have raised this in a discussion on the Next repo, but have received no answers as yet. Does anyone know what's causing this, please?

Cheers, Matt

**Comment by wongjn** [comment]
Do you have an example remotely-hosted git repository you could share please?

**Reply by MattHeslington** [reply]
Repo [here](https://github.com/MattHeslington/next-issue) - thanks for taking a look!

Codesandbox [here](https://codesandbox.io/p/github/MattHeslington/next-issue/draft/naughty-matsumoto?file=%2Fapp%2Fpage.tsx&selection=%5B%7B%22endColumn%22%3A46%2C%22endLineNumber%22%3A2%2C%22startColumn%22%3A46%2C%22startLineNumber%22%3A2%7D%5D&workspace=%257B%2522activeFileId%2522%253A%2522clawgh6za000e7og63lzu3izy%2522%252C%2522openFiles%2522%253A%255B%2522%252FREADME.md%2522%252C%2522%252Fapp%252Fpage.tsx%2522%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522spaces%2522%253A%257B%2522clawgh90t000r3f6c10aw7xti%2522%253A%257B%2522key%2522%253A%2522clawgh90t000r3f6c10aw7xti%2522%252C%2522name%2522%253A%2522Default%2520Space%2522%252C%2522devtools%2522%253A%255B%257B%2522key%2522%253A%2522clawgh90u000s3f6cgnkh6r53%2522%252C%2522type%2522%253A%2522PROJECT_SETUP%2522%252C%2522isMinimized%2522%253Afalse%257D%252C%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522key%2522%253A%2522clawghi99009g3f6cn7c98rce%2522%252C%2522isMinimized%2522%253Afalse%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522key%2522%253A%2522clawghgcr005l3f6c5cxtov25%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D%257D%252C%2522currentSpace%2522%253A%2522clawgh90t000r3f6c10aw7xti%2522%257D), but I'm not sure how useful it will be as it looks as though CS does a full refresh when a file is saved

**Reply by wongjn** [reply]
Seems like the HMR from Webpack does not see that the `global.css` needs to be HMR-ed when the `.tsx` file is changed. Editing the `global.css` file directly works however. It seems like possibly a bug, but I'm afraid I'm not clued in enough with the Webpack or Next.js internals to give a more helpful answer.
